pom.xml

```diff
@@ -45,8 +56,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>2.3.2</version>
                 <configuration>
-                    <source>1.6</source>
-                    <target>1.6</target>
+                    <source>1.8</source>
```

Comment:
For the exact changes I made, yes, 1.8 is required.  Attempting to build the project with 1.7 or 1.6 will result in compile-time errors:

Requirement for 1.7:

In the org.elasticsearch.transport.couchbase.capi package, the PluginSecurity.java file imports java.nio.file.Path, which appears to be provided only in 1.7+.

Requirement for 1.8:

In the org.elasticsearch.transport.couchbase.capi package, the CouchbaseCAPITransportImpl class contains a method called "doStart".  Inside of "doStart", I added a "doPrivileged" block, in which a variable "portNumber"  is referenced.  The declaration of "portNumber" occurs in the surrounding declaration of "onPortNumber".  In 1.7-, the compiler cannot accept a reference to a non-final variable inside of the "doPrivileged" block.

However, if the declaration "public boolean onPortNumber(int portNumber)" is changed to "public boolean onPortNumber(final int portNumber)", there is no longer a need for 1.8, at least for compilation purposes.

If anything above doesn't make sense, please let me know and I will try to clarify.  Thanks!
--
If it is desirable to keep the project at 1.7 (or even 1.6), please let me know.  

I can easily make the changes to get compatibility with 1.7, but I'd like to test the project again.  

To get compatibility with 1.6, at the very least, the java.nio libraries will have to be included as a dependency.  I'm not sure if java.nio is only 1.7+.  There may be alternatives to java.nio as well (I think I just copied most of the "PluginSecurity" file from another ES open-source plugin project, so I haven't spent much time on it.).  

Please let me know how you would like me to proceed.  Thanks!